### PR TITLE
kata-deploy: Build kata-agent as we build all the other components

### PR DESCRIPTION
**Pull request by fidencio**
For now we're not using the result of the build, but we should do it for the rootfs-build, and also for the tracing tests that will be migrated to GHA.

In order to test this PR, please, do, from the topdir of the kata-containers repo:
* `make agent-tarball`
* `make agent-opa-tarball`

### CODE REVIEW

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh [hunk 1]
```diff
@@ -106,6 +107,7 @@ docker run \
 	--env TARGET_BRANCH="${TARGET_BRANCH}" \
 	--env BUILDER_REGISTRY="${BUILDER_REGISTRY}" \
 	--env PUSH_TO_REGISTRY="${PUSH_TO_REGISTRY}" \
+	--env AGENT_CONTAINER_BUILDER="${INITRAMFS_CONTAINER_BUILDER}" \
```
Comment (stevenhorsman) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh:111]: I'm guessing this should be:
```
--env AGENT_CONTAINER_BUILDER="${AGENT_CONTAINER_BUILDER}" \
```

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh [hunk 2]
```diff
@@ -625,6 +628,32 @@ install_ovmf_sev() {
 	install_ovmf "sev" "edk2-sev.tar.gz"
 }
 
+install_agent_helper() {
+	agent_policy="${1:-no}"
+
+	latest_artefact="$(git log -1 --pretty=format:"%h" ${repo_root_dir}/src/agent)"
+	latest_builder_image="$(get_agent_image_name)"
+
+	install_cached_tarball_component \
```
Comment (stevenhorsman) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:637]: I might have missed something, but does this `agent_policy` value need to change the `install_cached_tarball_component` to differentiate between the cached agent with and without policy?

File: tools/packaging/static-build/agent/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2023 Intel
+#
+# SPDX-License-Identifier: Apache-2.0
+
+FROM alpine:3.18
+ARG GO_TOOLCHAIN
```
Comment (stevenhorsman) [inline tools/packaging/static-build/agent/Dockerfile:6]: nit: Is `GO_TOOLCHAIN` needed?

**Review by stevenhorsman — COMMENTED** [review]
Caching the agent will be super useful as it's the longest part of the kata build process in my experience, especially with the extra CoCo pieces. A few questions/comments from me:

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh (hunk 2 above)
Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:637]: We should just use the `build_target` as the component name, I think.

File: tools/packaging/static-build/agent/Dockerfile (hunk 3 above)
Comment (fidencio) [reply tools/packaging/static-build/agent/Dockerfile]: Nops. :-)

**Review by stevenhorsman — APPROVED** [review]
Thanks for the fixes. Code LGTM and I ran both commands locally and they worked

**Pull request merged**